feature/security/gnsi/certz/tests/client_certificates/client_certificates_test.go

```diff
@@ -0,0 +1,254 @@
+			mismatch:         false,
+		},
+		{
+			desc:             "Certz1.1:Load the correct key-type ecdsa trustbundle with 10CA configuration",
+			serverCertzFile:  "test_data/ca-10/server-ecdsa-a-cert.pem",
+			serverKeyzFile:   "test_data/ca-10/server-ecdsa-a-key.pem",
+			trustBundlezFile: "test_data/ca-10/trust_bundle_10_ecdsa_a.pem",
+			clientCertzFile:  "test_data/ca-10/client-ecdsa-a-cert.pem",
+			clientKeyzFile:   "test_data/ca-10/client-ecdsa-a-key.pem",
+			mismatch:         false,
+		},
+		{
+			desc:             "Certz1.1:Load the correct key-type rsa trustbundle with 1000CA configuration",
+			serverCertzFile:  "test_data/ca-1000/server-rsa-a-cert.pem",
+			serverKeyzFile:   "test_data/ca-1000/server-rsa-a-key.pem",
+			trustBundlezFile: "test_data/ca-1000/trust_bundle_1000_rsa_a.pem",
+			clientCertzFile:  "test_data/ca-1000/client-rsa-a-cert.pem",
+			clientKeyzFile:   "test_data/ca-1000/client-rsa-a-key.pem",
+			mismatch:         false,
+		},
+		{
+			desc:             "Certz1.1:Load the correct key-type ecdsa trustbundle with 1000CA configuration",
+			serverCertzFile:  "test_data/ca-1000/server-ecdsa-a-cert.pem",
+			serverKeyzFile:   "test_data/ca-1000/server-ecdsa-a-key.pem",
+			trustBundlezFile: "test_data/ca-1000/trust_bundle_1000_ecdsa_a.pem",
+			clientCertzFile:  "test_data/ca-1000/client-ecdsa-a-cert.pem",
+			clientKeyzFile:   "test_data/ca-1000/client-ecdsa-a-key.pem",
+			mismatch:         false,
+		},
+		{
+			desc:             "Certz1.2:Load the rsa trust_bundle from ca-02 with mismatching key type rsa client certificate from ca-01",
+			serverCertzFile:  "test_data/ca-01/server-rsa-a-cert.pem",
+			serverKeyzFile:   "test_data/ca-01/server-rsa-a-key.pem",
+			trustBundlezFile: "test_data/ca-02/trust_bundle_02_rsa_a.pem",
+			clientCertzFile:  "test_data/ca-01/client-rsa-a-cert.pem",
+			clientKeyzFile:   "test_data/ca-01/client-rsa-a-key.pem",
+			mismatch:         true,
+		},
+		{
+			desc:             "Certz1.2:Load the ecdsa trust_bundle from ca-02 with mismatching key type ecdsa client certificate from ca-01",
+			serverCertzFile:  "test_data/ca-01/server-ecdsa-a-cert.pem",
+			serverKeyzFile:   "test_data/ca-01/server-ecdsa-a-key.pem",
+			trustBundlezFile: "test_data/ca-02/trust_bundle_02_ecdsa_a.pem",
+			clientCertzFile:  "test_data/ca-01/client-ecdsa-a-cert.pem",
+			clientKeyzFile:   "test_data/ca-01/client-ecdsa-a-key.pem",
+			mismatch:         true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+
+			sc1, err := os.ReadFile(tc.serverCertzFile)
+			if err != nil {
+				t.Fatalf("Failed to read certificate: %v", err)
+			}
+			block, _ := pem.Decode(sc1)
+			if block == nil {
+				t.Fatalf("Failed to parse PEM block containing the public key.")
+			}
+			sCert, err := x509.ParseCertificate(block.Bytes)
+			if err != nil {
+				t.Fatalf("Failed to parse certificate: %v", err)
+			}
+
+			san := sCert.DNSNames[0]
```

Comment:
assuming [0] is the thing you want seems dangerous here.
I bet the testdata is setup to have this value, but also, what if it's not?

how would the production system determine if this was correct? (does it I guess? I don't know)

Reply:
There is one value set to the DNS in the server config file under test_data
subjectAltName=DNS:role001.pop55.net.example.com,URI:spiffe://teamname.groupname.location.place.example.com/ro
le/groupname-group

featureprofiles/feature/security/gnsi/certz/test_data# more server_cert_ext.cnf 
basicConstraints = CA:FALSE
nsCertType = server
nsComment = "Test Server Certificates - openconfig / featureprofiles"
subjectKeyIdentifier = hash
authorityKeyIdentifier = keyid,issuer:always
keyUsage = critical, digitalSignature, keyEncipherment
extendedKeyUsage = serverAuth
subjectAltName=DNS:role001.pop55.net.example.com,URI:spiffe://teamname.groupname.location.place.example.com/ro
le/groupname-group